lib/ZcapClient.js

```diff
@@ -0,0 +1,262 @@
+    }
+  }
+
+  /**
+   * Delegates an Authorization Capability to a target delegate.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} [options.url] - The relative URL to invoke the
+   *   Authorization Capability against, aka the `invocationTarget`. Either
+   *  `url` or `capability` must be specified.
+   * @param {string} [options.capability] - The parent capability to delegate.
+   *   Either `url` or `capability` must be specified.
+   * @param {string} options.targetDelegate - The URL identifying the entity to
+   *   delegate to.
+   * @param {string} [options.invocationTarget] - Optional invocation target
+   *   to use when narrowing a `capability`'s existing invocationTarget.
+   *   Default is to use `url` if `capability` is not provided, or
+   *   `capability.invocationTarget` if `capability` is provided.
+   * @param {string} [options.expires] - Optional expiration value for the
+   *   delegation. Default is 5 minutes after `Date.now()`.
+   * @param {string|Array} [options.allowedActions] - Optional list of allowed
+   *   actions or string specifying allowed delegated action. Default: [] -
+   *   delegate all actions.
+   *
+   * @returns {Promise<object>} - A promise that resolves to a delegated
+   *   capability.
+   */
+  async delegate({
+    url, capability, targetDelegate, invocationTarget, expires,
+    allowedActions = []
+  } = {}) {
+    let delegatedCapability;
+
+    // convert string value for allowedActions to array
+    allowedActions = (typeof allowedActions === 'string') ?
+      allowedActions = [allowedActions] : allowedActions;
+
+    // default expiration is 5 minutes in the future
+    const defaultExpires =
+      new Date(Date.now() + 5 * 60 * 1000).toISOString().slice(0, -5) + 'Z';
+
+    if(url) {
+      // generate the root capability
+      delegatedCapability = {
+        '@context': SECURITY_CONTEXT_V2_URL,
+        id: `urn:zcap:${await generateId()}`,
+        parentCapability: `urn:zcap:${url}`,
+        invocationTarget: url,
+        controller: targetDelegate,
+        expires: expires || defaultExpires
+      };
+    } else {
+      delegatedCapability = {
+        // use a provided capability
+        '@context': SECURITY_CONTEXT_V2_URL,
+        id: `urn:zcap:${await generateId()}`,
+        parentCapability: capability.id,
+        invocationTarget: invocationTarget || capability.invocationTarget,
+        controller: targetDelegate,
+        expires: expires || defaultExpires
+      };
+    }
+
+    if(allowedActions.length > 0) {
+      delegatedCapability.allowedActions = allowedActions;
+    }
+
+    const signedDelegatedCapability = await jsigs.sign(
+      delegatedCapability, {
+        suite: new Ed25519Signature2018({
+          signer: this.delegationSigner,
+          verificationMethod: this.delegationSigner.id
+        }),
+        purpose: new CapabilityDelegation({
+          capabilityChain: [
+            `urn:zcap:${url}`
```

Comment:
This is still not correct, it needs to:

1. If `url` was given, do what's here.
2. If `capability` was given, do not specify a `capabilityChain` it will be auto-computed (I believe ... this will need to be tested).

Reply:
Fixed in cb4c89137aa79130c922475c22109434f9fdb701.